test/src/test/java/org/corfudb/runtime/object/transactions/ImplicitTransactionsTest.java

```diff
@@ -0,0 +1,86 @@
+package org.corfudb.runtime.object.transactions;
+
+import com.google.common.reflect.TypeToken;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.collections.SMRMap;
+import org.corfudb.runtime.object.CorfuCompileProxy;
+import org.corfudb.runtime.object.ICorfuSMR;
+import org.corfudb.runtime.object.VersionLockedObject;
+import org.corfudb.runtime.view.ObjectsView;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Created by rmichoud on 7/28/17.
+ */
+public class ImplicitTransactionsTest extends AbstractTransactionsTest{
+
+    private CorfuCompileProxy getProxy(CorfuRuntime cr, String streamName) {
+        ObjectsView.ObjectID mapId = new ObjectsView.
+                ObjectID(CorfuRuntime.getStreamID(streamName), SMRMap.class);
+
+        CorfuCompileProxy cp = ((CorfuCompileProxy) ((ICorfuSMR) cr.getObjectsView().
+                getObjectCache().
+                get(mapId)).
+                getCorfuSMRProxy());
+        return cp;
+    }
+
+
+    @Override
+    public void TXBegin() {
+        OptimisticTXBegin();
+    }
+
+
+    /**
+     * Implicit transaction should not be nested. All implicit Tx
+     * end up in TXExecute, so here we execute isInNestedTransaction
+     * in TXExecute to check that we are indeed not in a nested tx.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void implicitTransactionalMethodAreNotNested() throws Exception {
+        CorfuRuntime rt = getDefaultRuntime().connect();
+        Map<String, String> m1 = rt.getObjectsView()
+                .build()
+                .setStreamName("test-1")
+                .setTypeToken(new TypeToken<SMRMap<String,String>>() {})
+                .open();
+
+
+        TXBegin();
+        m1.put("k1", "v1");
+
+        CorfuCompileProxy cp = getProxy(rt, "test-1");
+        boolean isInNestedTx = (boolean) cp.TXExecute(TransactionalContext::isInNestedTransaction);
```

Comment:
What do you mean by a custom object ?  A test Corfu Object that have an isInNestedTransaction @transactional method ? What is the benefit of doing it with a custom object instead of with the CorfuCompileProxy ?